osidb/models.py

```diff
@@ -112,7 +112,71 @@ class FlawType(models.TextChoices):
     WEAKNESS = "WEAKNESS"
 
 
-class FlawImpact(models.TextChoices):
+class ComparableTextChoices(models.TextChoices):
+    """
+    extension of the models.TextChoices classes
+    making them comparable with the standard operators
+
+    the comparison order is defined simply by the
+    top-down order in which the choices are written
+    """
+
+    @classmethod
+    def get_choices(cls):
+        """
+        get processed choices
+        """
+        return [choice[0] for choice in cls.choices]
+
+    @property
+    def weight(self):
+        """
+        weight of the instance for the comparison
+        defined by the order of the definition of the choices
+        """
+        return self.get_choices().index(str(self))
+
+    def _check_type(self, other):
+        """
+        ensure that that we are comparing the instances of the same type as
+        comparing different types (even two ComparableTextChoices) is undefined
+        """
+        if type(self) != type(other):
+            raise NotImplementedError(
+                f"Comparison of {type(self)} and {type(other)} is undefined"
+            )
```

Comment:
raising an error is a breach of contract for rich comparison methods, you need to return a boolean value or `NotImplemented` (which is different from `NotImplementedError`

Reply:
I consider this as a programmer error and reacts on your comment

> Yes, comparing ComparableTextChoices to other is a programming error, but the error being raised is not a correct one: it should tell the programmer that he's comparing apples to oranges, not that attribute x is not a member of y as that is a confusing error.

telling the programmer what exactly (s)he did wrong. Why not to be vocal about it?